src/main/java/dk/dbc/updateservice/update/NoteAndSubjectExtensionsHandler.java

```diff
@@ -218,6 +275,149 @@ MarcRecord recordDataForRawRepo(MarcRecord marcRecord, String groupId) throws Up
         return result;
     }
 
+    private void validateCatalogCodes(MarcRecord record) throws UpdateException {
+        final List<MarcField> newCatalogCodeFields = record.getFields().stream()
+                .filter(field -> field.getName().matches(CATALOGUE_CODE_FIELD)).collect(Collectors.toList());
+        for (MarcField field : newCatalogCodeFields) {
+            for (MarcSubField subField : field.getSubfields()) {
+                if (subField.getName().equals("a")) {
+                    final String msg = messages.getString("update.library.record.catalog.codes.x.only");
+                    LOGGER.error("Unable to create sub actions due to an error: {}", msg);
+                    throw new UpdateException(msg);
+                }
+            }
+        }
+
+    }
+
+    private String getOveCode(MarcField newCatalogField) {
+        for (MarcSubField subField : newCatalogField.getSubfields()) {
+            if ("x".equals(subField.getName()) && subField.getValue().startsWith("OVE")) return subField.getValue();
+        }
+        return "";
+    }
+
+    List<MarcSubField> updateOveAndAmp(MarcField field, String groupId, String oveCode) {
+        List<MarcSubField> newList = new ArrayList<>();
+        List<MarcSubField> subs = field.getSubfields();
+        for (MarcSubField s1 : subs) {
+            if ("&".equals(s1.getName())) continue;
+            if ("x".equals(s1.getName()) && s1.getValue().startsWith("OVE")) continue;
```

Comment:
Det bør være if-sætning som checker betingelserne er opfyldt, i stedet for at skippe hvis betingelserne ikke er opfyldt.

```
        for (MarcSubField s1 : subs) {
            if (!"&".equals(s1.getName()) 
                    && !"x".equals(s1.getName()) && s1.getValue().startsWith("OVE")) {
                newList.add(s1);
            }
        }
```